src/operator/nn/dropout-inl.h

```diff
@@ -94,47 +72,124 @@ struct DropoutParam : public dmlc::Parameter<DropoutParam> {
 
 template<typename xpu, typename DType>
 class DropoutOp : public Operator {
+#if defined(USE_MKL) && defined(_OPENMP)
+  static void BernoulliGenerate(common::random::RandGenerator<cpu, DType> gen,
+                                int n, double p, int* r) {
+    typename RandGenerator<xpu, DType>::Impl genImpl(&gen, 1);
+    const int seed = 17 + genImpl.rand() % 4096;  // NOLINT(runtime/threadsafe_fn)
+    const int nthr = engine::OpenMP::Get()->GetRecommendedOMPThreadCount();
+#pragma omp parallel num_threads(nthr)
+    {
+      const int ithr = omp_get_thread_num();
+      const int avg_amount = (n + nthr - 1) / nthr;
+      const int my_offset = ithr * avg_amount;
+      const int my_amount = std::min(my_offset + avg_amount, n) - my_offset;
+      if (my_amount > 0) {
+        VSLStreamStatePtr stream;
+        vslNewStream(&stream, VSL_BRNG_MCG31, seed + my_offset);
+        vslSkipAheadStream(stream, my_offset);
```

Comment:
can we move this to Resource::ParallelRandom so that all rngs are fast when using mkl?

Reply:
I think parallel random can be upgraded to use viRngUniform (for instance), but I think that's for a different PR. It'd be a big change.

Reply:
+1. We should make MKL support a first class citizen

Reply:
Note that this is RngBernoulli, which isn't covered by ParallelRandom

Reply:
@cjolivier01 OK. we can merge this first.

but please make sure we track it somewhere.